src/main/java/com/lukaspradel/steamapi/webapi/request/SteamWebApiRequestHandler.java

```diff
@@ -82,33 +86,26 @@ URI getRequestUri(String scheme, String host, String path, List<NameValuePair> p
 
 	String getWebApiResponse(URI requestUrl) throws SteamApiException {
 		HttpClient client = getHttpClient();
-		HttpGet getRequest = new HttpGet(requestUrl);
-
-		// this try-with-resources statement closes the connection
-		// no need to close the client (ref https://hc.apache.org/httpcomponents-client-4.5.x/quickstart.html)
-		try (ClassicHttpResponse response = client.executeOpen(null, getRequest, null)) {
-			Integer statusCode = response.getCode();
-
-			if (!statusCode.toString().startsWith("20")) {
-				if (statusCode.equals(HttpStatus.SC_UNAUTHORIZED)) {
-					throw new SteamApiException(
-							SteamApiException.Cause.FORBIDDEN, statusCode, response.getReasonPhrase());
-				}
-				throw new SteamApiException(
-						SteamApiException.Cause.HTTP_ERROR, statusCode, response.getReasonPhrase());
-			}
-			return getHttpResponseAsString(response);
-		} catch (IOException | ParseException e) {
+		HttpRequest getRequest = HttpRequest.newBuilder(requestUrl).build();
+
+		try {
+			var response = client.send(getRequest, BodyHandlers.ofString());
+			int statusCode = response.statusCode();
+
+			if (Integer.toString(statusCode).startsWith("20"))

```

Comment:
Compiler will likely optimize this anyway but for clarity I would prefer to change this into
```
if () {
} else if {
} else {
}
```

if it's ok with you